GarNet/garnet.py

```diff
@@ -226,23 +153,34 @@ def TF_regression(motifs_and_genes_file_or_dataframe, expression_file, output_di
 		# Occasionally there's only one gene associated with a TF, which we can't fit a line to.
 		if len(expression_profile) < 5: continue
 
+		# This allows heavier points to be visualized on the top instead of being hidden by long distance points 
+		# expression_profile = expression_profile.reindex(expression_profile.motif_gene_distance.abs().sort_values(inplace=False, ascending=False).index)
+
 		# Ordinary Least Squares linear regression
 		result = linear_regression(formula="expression ~ motifScore", data=expression_profile).fit()
 
 		if output_dir:
 			plot = plot_regression(model_results=result, ax=expression_profile.plot(x="motifScore", y="expression", kind="scatter", grid=True))
+			# Add color to points based on distance to gene
+			plt.scatter(expression_profile["motifScore"], expression_profile["expression"],
+				c=[abs(v) for v in expression_profile["motif_gene_distance"].tolist()],
+				norm=matplotlib.colors.LogNorm(vmin=1, vmax=100000, clip=True), 
+				cmap=matplotlib.cm.Blues_r)
+			plt.title("%s, %0.4f" %(TF_name, result.pvalues['motifScore']))
+			plt.colorbar()
+
 			os.makedirs(os.path.join(output_dir, "regression_plots"), exist_ok=True)
-			plot.savefig(os.path.join(output_dir, "regression_plots", TF_name + '.jpg'))
+			plot.savefig(os.path.join(output_dir, "regression_plots", TF_name.replace("/", "-") + '.pdf'))
```

Comment:
What was happening here? Were there TFs with "/" in them causing issues?

Reply:
Yes, this was when we were using motifMap which still contained non gene symbols.